### ISSUE TITLE
PS3 DS3 Model CECHZC2U

### ISSUE THREAD

**Issue by fancyf** [digitalLumberjack/recalbox-os#394]
Hey!

ps3_sixad_version = CECHZC2U

When does this config entry gets implemented? :)

**Comment by Fuhrmann** [comment]
Same. I have the same model (Gasia) and it's not working. Even connected via USB. The controller works fine (tested in a PS3). Thanks guys.

**Comment by digitalLumberjack** [comment]
Are you on the beta ? Did you read the PS3 controller wiki ? 
https://github.com/digitalLumberjack/recalbox-os/wiki/PS3-controllers-drivers-%28EN%29

**Comment by fancyf** [comment]
Yes, already tried with "shanwan" and "gasia" in 3.3 beta config. No luck.

**Comment by digitalLumberjack** [comment]
With a reboot ?

**Comment by fancyf** [comment]
yep sure :)

**Comment by digitalLumberjack** [comment]
Hum. Weird. A friend of mine may have gasia, i'll get it.

**Comment by digitalLumberjack** [comment]
If it's new model, we may have a problem...

**Comment by fancyf** [comment]
if you need any controller data etc. let me know... thx!

**Comment by fancyf** [comment]
Sorry for pushing the thread, but I bought 4 of these controllers only for recalbox. I dont have a playstation :) If it doesnt work with em I need to send them back this week. Do you think they will work or should i rather buy original controllers?

**Comment by johanpoirier** [comment]
Same problem for me here

**Comment by johanpoirier** [comment]
I have two of these controllers waiting for nothing. Will they work with the RecalBox someday?

Thanks.

**Comment by caverna** [comment]
Testing with 3.3.0-beta-9 without success...

**Comment by johanpoirier** [comment]
With the latest beta and the new line in the config file, it works for me:
```
ps3_sixad_version = CECHZC2U
```

**Comment by digitalLumberjack** [comment]
uber weird @johanpoirier as the recalbox.conf of the 3.3.0 beta accept only `controllers.ps3.driver`

**Comment by johanpoirier** [comment]
@digitalLumberjack I will post my config file as soon as I get home.

**Comment by johanpoirier** [comment]
I updated to the 3.3.0-beta-9 this morning.

Here is my config file:
```
## Please enable only one of these
# -------------- D1 - PS3 Controllers ------------ #
##Enable PS3 controllers support
controllers.ps3.enabled=1
## Choose an driver beetween official, shanwan and gasia if you have dualshock clones (official,shanwan,gasia)
controllers.ps3.driver=shanwan
ps3_sixad_version = CECHZC2U
```

My controller:
![img_20151013_093543-2](https://cloud.githubusercontent.com/assets/142127/10448734/037ea2bc-718f-11e5-84cd-ed7c10f05d61.jpg)

It is synced to the recalbox:
![img_20151013_093554](https://cloud.githubusercontent.com/assets/142127/10448735/03985400-718f-11e5-95d2-f0c7dad9fe0f.jpg)

**Comment by caverna** [comment]
From my side, I didn't have success, but I notice something that may be a clue...
My joystick has the text "dualshock 3 and sixaxis" at right of usb connector, not at the bottom.
The bottom label is a little different from yours, but it's written 'made in China' too.
If I use shanwan or gasia drivers, nothing happens, but when I use official and press PS button, things happens:
- If recalbox is in screen save mode, it wakes up;
- The joystick blinks like spected before pairing;
- The lights turn on in sequence 1-2-3-4-3-2-1-2...etc;
- At the same time, it rumbles for almost 4 seconds;
- In the end, the 4 lights blinks and stay off;
I have 2 joysticks and it happen with both;
One more thing, just one of them, when I plug the usb cable, I see a color square at the top right screen and the power led in RPi turns off for a moment. It keeps happening while the joystick is connected (before someone ask, it didn't sync with leds charging frequence)

**Comment by yusf** [comment]
@caverna That top right color square is a power surge indicator saying your power supply needs to be replaced with at stronger one. Look for one with at least 2A.

**Comment by c4store82** [comment]
I have a GASIA controller.  I am having trouble changing my PS3 driver on recalbox.conf
What exactly do I have to type to switch it?

**Comment by substring** [comment]
@c4store82 that's no issue (yet). Please read the wiki link above, or post in the forum

**Comment by francoiskha** [comment]
config file : 
```
# -------------- D1 - PS3 Controllers ------------ #
##Enable PS3 controllers support
controllers.ps3.enabled=1
## Choose an driver beetween official, shanwan and gasia if you have dualshock clones (official,shanwan,gasia)
controllers.ps3.driver=shanwan
```
works on my Rpi2, with recalbox 4.0.0-beta2, a shanwan controller CECHZC2U (commercial brand is Althemax), and a bluetooth dongle `CSR8510 A10` (commercial name is CSL - USB Adaptateur Bluetooth V4.0).
There is a user experience problem though :

- When I pair the controller via the user interface, it vibrates and stop blinking, BUT recalbox tells me there is no controller found. Via `hcitool scan`, the controller acts the same way and there is nothing in console output. There is a lack of information somewhere because in both cases, there is a line in dmesg like
```
[   50.781655] input: PLAYSTATION(R)3 Controller (03:8E:88:67:98:54) as /devices/virtual/input/input1
```
and I can configure the controller and play.

- I have to remake the pairing and the controller config after each reboot.

**Comment by expouic** [comment]
Hi guys!
When connect my "fake" ps3 controller and execute the command `dmesg` have this error!

```
> [ 1035.469151] sony 0003:054C:0268.0016: usb_submit_urb(ctrl) failed: -1
> [ 1035.469176] sony 0003:054C:0268.0016: timeout initializing reports
> [ 1035.469437] input: ShanWan PS(R) Ga`epad as /devices/platform/soc/3f980000.usb/usb1/1-1/1-1.2/1-1.2:1.0/0003:054C:0268.0016/input/input20
> [ 1035.469901] sony 0003:054C:0268.0016: input,hiddev0,hidraw1: USB HID v1.10 Joystick [ShanWan PS(R) Ga`epad] on usb-3f980000.usb-1.2/input0
> [ 1036.646237] usb 1-1.2: USB disconnect, device number 28
> [ 1036.879145] usb 1-1.2: new full-speed USB device number 29 using dwc_otg
> [ 1036.985750] usb 1-1.2: New USB device found, idVendor=054c, idProduct=0268
> [ 1036.985759] usb 1-1.2: New USB device strings: Mfr=1, Product=2, SerialNumber=0
> [ 1036.985765] usb 1-1.2: Product: PS(R) Ga`epad
> [ 1036.985772] usb 1-1.2: Manufacturer: ShanWan
```

and the controller not pair :/

any ideia to resolve that issue?

**Comment by mcapuccini** [comment]
@johanpoirier I have your same controller, but no luck with recalbox 4.0.1 and 4.0.2. What beta did you use to make it work?

**Comment by mcapuccini** [comment]
never mind... the problem was that I was using a Raspberry PI 3, with 2B it works as a charm
